backend/data_adapter.go

```diff
@@ -17,7 +19,33 @@ func newDataSDKAdapter(handler QueryDataHandler) *dataSDKAdapter {
 	}
 }
 
+const (
+	authHeader     = "Authorization"
+	xIDTokenHeader = "X-ID-Token"
+)
+
+func withOAuthMiddleware(ctx context.Context, authorization, xIDToken string) context.Context {
+	if authorization != "" {
+		ctx = httpclient.WithContextualMiddleware(ctx,
+			httpclient.MiddlewareFunc(func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+				return httpclient.RoundTripperFunc(func(qreq *http.Request) (*http.Response, error) {
+					// Only set the Authorization header if it is not already set.
+					if qreq.Header.Get(authHeader) == "" {
+						qreq.Header.Set(authHeader, authorization)
+					}
+					// Only set the X-ID-Token header if it is not already set.
+					if xIDToken != "" && qreq.Header.Get(xIDTokenHeader) == "" {
+						qreq.Header.Set(xIDTokenHeader, xIDToken)
+					}
+					return next.RoundTrip(qreq)
+				})
+			}))
+	}
+	return ctx
+}
+
 func (a *dataSDKAdapter) QueryData(ctx context.Context, req *pluginv2.QueryDataRequest) (*pluginv2.QueryDataResponse, error) {
+	ctx = withOAuthMiddleware(ctx, req.Headers[authHeader], req.Headers[xIDTokenHeader])
```

Comment:
Quick feedback. I would suggest forwarding all http headers here rather than specific ones since it's already been handled in core Grafana. 

Maybe something like this?

```suggestion
	sReq := FromProto().QueryDataRequest(req)
	headers := sReq.GetHTTPHeaders()
	withMiddleware(..., headers)
```

Reply:
Sure, I don't mind but have a question:

> I would suggest forwarding all http headers here rather than specific ones since it's already been

you mean that core Grafana already strip "unsupported" headers so it's not necessary to check that here in the SDK or that for core plugins already forward all http headers? I don't see code that does any of those things but I may not know where to look.

Reply:
no need to handle separately cookies now so :thumbsup:

Reply:
https://github.com/grafana/grafana/tree/main/pkg/services/pluginsintegration/clientmiddleware and https://github.com/grafana/grafana/blob/ed626427de8da5895a24bd7c6aef81e24b663386/pkg/plugins/manager/client/client.go#L93